### PR TITLE
Rework list screen UI and add infinite scroll

### LINKED ISSUE #100: Rework list screen UI and make infinite scroll
### Problem
At this stage, the list screen only loads 5 nearest parkings rather than dynamically loading more and more parkings as we scroll.

- Add fetch functionality when we reach bottom of scroll using lazylist

Also rework ui of the screen

### PR DESCRIPTION
**Pull request by Armetiis**
Closes #100 

### What
This PR does two things:
1. Rework the UI of the list screen, here is a preview
image
image

2. Make the list screen infinite scroll

### Key changes
For consistency while scrolling, the filtering system had to be changed. We are back to a classic filtering system, as the smart filtering system and the infinite scroll are not compatible. (Previously, the parkings were first sorted by number of matched criterion and then by distance. This would mean that while scrolling down, when we would reach the bottom, parkings that were further but had more matched criterion would get propulsed to the top of the list.)

### Coverage 
image

### Tests
Tests have been adapted to cope with the new UI.

### Future work:
Once the StarReview molecule is done, it shall be added in place of the text "Rating:"
Once the logic for the image database is done, we can add an image to preview the parking

### CODE REVIEW

File: app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt [hunk 1]
```diff
@@ -193,6 +192,36 @@ fun ToggleButton(
       }
 }
 
+/**
+ * Create a themed toggle button. This overload offers more flexibility.
+ *
```
Comment (andrepeiry) [inline app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt:197]: Missing  `@param text`

File: app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt [hunk 2]
```diff
@@ -193,6 +192,36 @@ fun ToggleButton(
       }
 }
```
Comment (andrepeiry) [inline app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt:193]: Overload + modify an unused atom seems a bit overkill, maybe it would be simpler to tune the atom to your needs.

Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt:193]: Tried to. The two are not really compatible.

File: app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt [hunk 3]
```diff
@@ -239,83 +243,103 @@ fun FilterSection(
...
-      content()
-    }
-  }
+        if (isExpanded) {
+          content()
+        }
+      }
 }
 
+@SuppressLint("DefaultLocale")
```
Comment (andrepeiry) [inline app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt:265]: I'm not sure that explicitly removing a warning is a good thing, moreover when Android Studio is telling you "*Implicitly using the default locale is a common source of bugs: Use `String.format(Locale, ...)` instead*".

File: app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt (hunk 2 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt:193]: I've looked it at multilple angles and this is what makes the most sense.

File: app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt (hunk 3 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/list/ListScreen.kt]: Sometimes you kind of have to suppress warnings, but in this case since Android studio explicitly suggests an alternative it might be good to use it just in case

File: app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt [hunk 4]
```diff
@@ -33,11 +33,6 @@ class ParkingViewModel(
   private val _selectedParking = MutableStateFlow<Parking?>(null)
   val selectedParking: StateFlow<Parking?> = _selectedParking
 
-  // TODO: Replace with actual location
```
Comment (LouisJamesVasseur) [inline app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt:36]: Why is this removed? Is this linked to the feature you're implementing in this PR?

File: app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt (hunk 2 above)
Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/ui/theme/atoms/Button.kt:195]: On the principle I agree w/ @andrepeiry but I think that the way that the code is implemented here works just fine

**Comment by LouisJamesVasseur** [comment]
Great work, the screen and the code both look great!

**Review by LouisJamesVasseur — APPROVED** [review]
Great work, the screen and the code both look great!

File: app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt (hunk 4 above)
Comment (Armetiis) [reply app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt:36]: Yes. I added a LaunchedEffect in the ListScreen. There's no reason between creating the viewmodel and loading parkings. The loading of the k closest parkings should be handled by the screen that uses it

Comment (LouisJamesVasseur) [reply app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt:36]: Makes sense, thanks for the explanation

**Review by andrepeiry — APPROVED** [review]
Everything seems to be working, nice work ^^

**Review by andrepeiry — APPROVED** [review]
